dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -980,64 +981,85 @@ def _DownloadResults(self,
       results.append(data_frame)
     return results
 
+  def _ProcessQuery(
+      self,
+      hostname: str,
+      osquery_container: containers.OsqueryQuery
+  ) -> None:
+    """Processes an osquery flow on a GRR client.
+
+    Args:
+      hostname (str): the GRR client hostname.
+      osquery_container (containers.OsqueryQuery): the osquery.
+    """
+    client = self._GetClientBySelector(hostname)
+
+    hunt_args = osquery_flows.OsqueryFlowArgs(
+        query=osquery_container.query,
+        timeout_millis=self.timeout_millis,
+        ignore_stderr_errors=self.ignore_stderr_errors)
+
+    try:
+      flow_id = self._LaunchFlow(client, 'OsqueryFlow', hunt_args)
+      self._AwaitFlow(client, flow_id)
+    except DFTimewolfError as error:
+      self.ModuleError(
+          f'Error raised while launching/awaiting flow: {error.message}')
+      return
```

Comment:
it would be interesting to also account for reporting when a flow gets canceled because of a timeout (given that we provide the timeout threshold in this function). (I suspect this would mess with the rest of your code here)

Reply:
Agree on both - it would be interesting and imo it would make the function flow more confusing (pun not intended).  For now, I would opt to handle flow timeouts more generically in the GRRFlow module (_AwaitFlow most likely).  I've created https://github.com/log2timeline/dftimewolf/issues/698 to look into this.